### DISCUSSION TITLE
Zendesk Hook doesn't support token based authentication

### DISCUSSION THREAD

**Discussion by sharathsamala** [apache/airflow#22654] [General] [answered]
https://github.com/apache/airflow/blob/f41416744f0f55fba89f647e1b52c881786ecf33/airflow/providers/zendesk/hooks/zendesk.py#L63

Hi Guys, i'm trying to use zendesk hook in one of our airflow dag. I have setup the zendesk connection using airflow connections module. Our zendesk authentication is based on the token but not the password. I have provided the token in extras section of the connection. 

When i saw the source code, i came to know that zendesk hook is not having the ability to use the token. Please let me know if i'm missing something. I would like to contribute to this issue

**Comment by josh-fell** [comment, marked as answer]
From a cursory look around, you are correct. I think you should absolutely submit a PR to make use of the token auth if you are up for it. There are over 2K contributors to Airflow and we definitely encourage folks to contribute if they would like to!

As a workaround for now, you can subclass the `ZendeskHook` and update its `_init_conn()` method to fit your needs.
